cumulusci/core/template_utils.py

```diff
@@ -47,9 +48,11 @@ def __getattr__(self, name):
 
 faker_template_library = FakerTemplateLibrary()
 
+Template = lru_cache(512)(Template)
+
 
-def format_str(value, i):
-    if isinstance(value, str):
-        value = Template(value).render(number=i, fake=faker_template_library)
+def format_str(value, variables={}, fake=faker_template_library):
```

Comment:
Don't use mutable defaults! It's a trap!

Reply:
Good catch!

Reply:
Although it is a trap, in this case the function doesn't (and shouldn't) mutate the dict. In the future I would use a frozenmap as the default to enforce that.